pkg/cli/admin/upgrade/recommend/recommend.go

```diff
@@ -0,0 +1,236 @@
+// Package recommend displays recommended update information.
+package recommend
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/blang/semver"
+	"github.com/spf13/cobra"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericiooptions"
+	kcmdutil "k8s.io/kubectl/pkg/cmd/util"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configv1client "github.com/openshift/client-go/config/clientset/versioned"
+)
+
+const (
+	// clusterStatusFailing is set on the ClusterVersion status when a cluster
+	// cannot reach the desired state. It is considered more serious than Degraded
+	// and indicates the cluster is not healthy.
+	clusterStatusFailing = configv1.ClusterStatusConditionType("Failing")
+)
+
+func newOptions(streams genericiooptions.IOStreams) *options {
+	return &options{
+		IOStreams: streams,
+	}
+}
+
+func New(f kcmdutil.Factory, streams genericiooptions.IOStreams) *cobra.Command {
+	o := newOptions(streams)
+	cmd := &cobra.Command{
+		Use:   "recommend",
+		Short: "Recommends cluster updates.",
```

Comment:
nit: Maybe `Displays cluster update recommendations.`  or something else that includes "display" or other wording to underline the read-only-ness?  Maybe also additional sentences like:

> This subcommand is read-only, and does not affect the state of the cluster.  To initiate an update, use the 'oc adm upgrade ...' subcommand.

Reply:
Thank you for the suggestion. `Displays cluster update recommendations.` looks better IMO.

I have also added a longer description.